src/adera/Shaders/PlumeShader.h

```diff
@@ -58,12 +63,13 @@ class PlumeShader : public Magnum::GL::AbstractShaderProgram
 
     PlumeShader();
 
-    static void draw_plume(osp::active::ActiveEnt e,
-        osp::active::ActiveScene& rScene,
-        osp::active::ACompCamera const& camera) noexcept;
+    static void draw_plume(
+            osp::active::ActiveEnt ent, osp::active::ActiveScene& rScene,
+            osp::active::ACompCamera const& camera, void* pUserData) noexcept;
```

Comment:
Can't say i'm a fan of passing around void*.

Is this absolutely unavoidable?

Reply:
Some kind of non-owning `any` can help add some type checking. This part of the code may be hot, but a bit of extra overhead probably wouldn't be the bottleneck.

Reply:
Shouldn't be that hot, since it'll only be called at most (number of thrusters onscreen) times per frame.

Actually, what am I saying. If KSP is any indication, that number might be pretty large.

In any case, at that stage it's entirely possible that much of this framework will have been rewritten for higher performance, so I wouldn't worry about it too much. If you've ever seen one of those "frame breakdown" videos from graphics-hungry games, they're always absurdly complicated, so it's safe to assume our renderer will undergo a lot of evolution over time, and the notion of drawing individual objects via a shader will likely no longer exist.